<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml"><!-- InstanceBegin template="/Templates/VarScan.dwt" codeOutsideHTMLIsLocked="false" -->
<head>
<!-- InstanceBeginEditable name="doctitle" -->
<title>VarScan - Variant Detection in Massively Parallel Sequencing Data - Support FAQ</title>
<script type="text/javascript">

  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', 'UA-33985908-1']);
  _gaq.push(['_trackPageview']);

  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();

</script>
<!-- InstanceEndEditable --><meta http-equiv="Content-Language" content="English" />
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
<link rel="stylesheet" type="text/css" href="style.css" media="screen" />
<!-- InstanceBeginEditable name="head" --><!-- InstanceEndEditable -->
</head>
<body>

<div id="wrap">

<div id="header">
<TABLE CELLSPACING=0 CELLPADDING=0 BORDER=0>
<TR><TD WIDTH="100%">
<h1><a href="index.html" TITLE="VarScan variant detection tool">VarScan</a></h1>
<h2>variant detection in massively parallel sequencing data</h2></TD>
<TD valign="bottom"><A HREF="https://varscan.sourceforge.net" TITLE="VarScan Variant Calling Software"><IMG SRC="images/VarScan-FancyLogo.jpg" ALT="VarScan variant detection software" BORDER=0></A></TD>
</TR>
</TABLE>
</div>

<div id="menu">
<ul>
<li><a href="index.html" title="VarScan snp and indel caller">Home</a></li>
<li><a href="https://sourceforge.net/projects/varscan/" TITLE="Variant Discovery Project">Project</a></li>
<li><a href="https://sourceforge.net/projects/varscan/files/" TITLE="Download Mutation Detection Software">Download</a></li>
<li><a href="using-varscan.html" TITLE="Variant calling documentation">Manual</a></li>
<li><a href="support-faq.html" TITLE="Variant detection FAQ">FAQ</a></li>
<li><a href="doc/index.html" TITLE="VarScan documentation">Documentation</a></li>
<li><a href="https://sourceforge.net/projects/varscan/forums/forum/1073559" TITLE="VarScan Support">Support Forum</a></li>
</ul>
</div>

<div id="content">

<div class="left"> 

<h2>VarScan</h2>
<ul>
<li><a href="https://sourceforge.net/apps/mediawiki/varscan" TITLE="VarScan software">About/Wiki</a></li> 
<li><a href="https://sourceforge.net/projects/varscan/" TITLE="Mutation Discovery Project">Project Page</a></li> 
<li><a href="using-varscan.html" TITLE="Variant calling documentation">User's Manual</a></li> 
<li><a href="doc/index.html" TITLE="VarScan Javadoc">JavaDoc</a></li> 
<li><a href="https://sourceforge.net/projects/varscan/files/" TITLE="Download variant caller software" STYLE="font-weight: bold; color:#00CC00">Download Now</a></li>
<li><a href="support-faq.html" TITLE="Variant caller support FAQ">Support/FAQ</a></li>
<li><a style="color: #FF0000" href="https://sourceforge.net/projects/varscan/forums/forum/1073559" TITLE="VarScan Support">Support Forum</a></li>
</ul>

<h2>Calling Methods</h2>
<ul>
<li><a href="germline-calling.html" TITLE="Germline variant calling methods">Germline Variants</a></li> 
<li><a href="somatic-calling.html" TITLE="Somatic mutation calling methods">Somatic Mutations</a></li> 
<li><a href="copy-number-calling.html" TITLE="Somatic copy number alteration calling methods">Copy Number Variants</a></li> 
<li><a href="trio-calling-de-novo-mutations.html" TITLE="de novo mutations from trio calling"><I>de novo</I> Mutations (Trios)</a></li> 
</ul>

<h2>WashU Analysis Tools</h2>
<ul>
<li><a href="http://gmt.genome.wustl.edu" TITLE="Genome Modeling Tools">Genome Modeling Tools</a></li> 
<li><a href="http://gmt.genome.wustl.edu/breakdancer/1.2/index.html" TITLE="Breakdancer">BreakDancer</a> (SVs)</li> 
<li><a href="http://gmt.genome.wustl.edu/somatic-sniper/0.7.4/index.html" TITLE="Somatic Sniper">Somatic Sniper</a> (SNVs)</li> 
<li><a href="http://cmds.sourceforge.net/" TITLE="CMDS">CMDS</a> (Copy Number)</li> 
</ul>


<h2>SAM/BAM Format</h2>
<ul>
<li><a href="http://samtools.sourceforge.net/SAM1.pdf" TITLE="SAM Format" TARGET="_blank">SAM Format Spec</a></li> 
<li><a href="http://samtools.sourceforge.net/pileup.shtml" TITLE="SAMtools Pileup Format" TARGET="_blank">Pileup Format</a></li> 
<li><a href="http://samtools.sourceforge.net" TITLE="SAMtools" TARGET="_blank">SAMtools</a></li> 
<li><a href="http://picard.sourceforge.net/index.shtml" TITLE="Picard Tools" TARGET="_blank">Picard</a></li> 
</ul>

<h2>Short Read Aligners</h2>
<ul>
<li><a href="http://bowtie-bio.sourceforge.net" title="bowtie aligner" TARGET="_blank">Bowtie</a></li> 
<li><a href="http://bio-bwa.sourceforge.net" title="bwa aligner" TARGET="_blank">BWA</a></li> 
<li><a href="http://maq.sourceforge.net" title="maq aligner" TARGET="_blank">Maq</a></li> 
<li><a href="http://www.novocraft.com" title="novocraft aligner" TARGET="_blank">Novoalign</a></li> 
<li><a href="http://www.sanger.ac.uk/resources/software/ssaha2" title="ssaha aligner" TARGET="_blank">SSAHA2</a></li> 
</ul>


</div>


<div class="right"> 
<!-- InstanceBeginEditable name="MainBody" -->

<h2><a href="#">Support FAQ</a></h2>
This page contains answers to many of the common questions asked about VarScan usage, performance, input/output, etc.

<BR><BR>
<H3>USAGE QUESTIONS</H3>
<A HREF="#usage-which-version"><B>Which version of VarScan should I use?</B></A><BR>
<A HREF="#usage-which-command"><B>Which VarScan command should I use?</B></A><BR>
<A HREF="#usage-wgs-exome-rna"><B>Can I use VarScan on WGS, exome, or RNA-seq data?</B></A><BR>
<A HREF="#usage-pooled-samples"><B>Does VarScan work for pooled samples?</B></A><BR>
<A HREF="#usage-validation"><B>How do I use VarScan for validation?</B></A><BR>
<A HREF="#usage-model-microbial"><B>Can I use VarScan on my model organism or microbial genome?</B></A><BR>
<BR>

<H3>INPUT QUESTIONS</H3>
<A HREF="#input-what-format"><B>What input format should I give to VarScan?</B></A><BR>
<A HREF="#input-aligner-choice"><B>Which aligner should I use?</B></A><BR>
<A HREF="#input-base-qualities"><B>Do I need Illumina or Phred scale base qualities?</B></A><BR>
<A HREF="#input-mapping-qualities"><B>What about mapping qualities?</B></A><BR>
<BR>

<H3>OUTPUT QUESTIONS</H3>
<A HREF="#output-column-meanings"><B>What do the output columns mean?</B></A><BR>

<A HREF="#output-p-value"><B>How are the p-values calculated?</B></A><BR>

<A HREF="#output-p-value-98"><B>Why are all of my p-values 0.98?</B></A><BR>

<A HREF="#output-filtering"><B>How should I filter the output files?</B></A><BR>

<A HREF="#output-confidence"><B>Does VarScan provide a confidence score similar to SAMtools?</B></A><BR>

<BR />
<H3>COMMON ISSUES, WARNINGS, AND ERRORS</H3>

<A HREF="#error-resetting-file"><B>Warnings about "resetting normal" or "resetting tumor" file</B></A><BR />

<A HREF="#read-counts-different"><B>Read counts from VarScan are different from SAMtools/IGV counts</B></A><BR />

<BR><BR>
<H3>USAGE QUESTIONS</H3>

<H4><A NAME="usage-which-version"><font CLASS="likeH3"><B>Which version of VarScan should I use?</B></font></A></H4>

Always use the latest version! Currently, this is v2.3.x. There are differences between versions; notably, between 
v2.2.2 and v2.2.3 we adjusted how reads are counted for indels, which had exhibited strange behavior due to some 
peculiarities in how SAMtools represents indels in the pileup files.

<BR><BR>


<H4><A NAME="usage-which-command"><font CLASS="likeH3"><B>Which VarScan command should I use?</B></font></A></H4>

If you have a single sample and wish to call variants, you can use <I>pileup2snp</I> to call SNPs, <I>pileup2indel</I> to call indels, 
or <I>pileup2cns</I> to call consensus genotypes at every position meeting the coverage requirement. 
To call both SNPs and indels simultaneously, use <I>pileup2cns</I> with the <I>--variants</I> parameter set to 1.
If you have tumor-normal pairs, you should use the <I>somatic</I> command to call mutations (SNVs/indels) and the 
<I>copynumber</I> command to call copy number alterations (CNAs).

<BR><BR>


<H4><A NAME="#sage-wgs-exome-rna"><font CLASS="likeH3"><B>Can I use VarScan on WGS, exome, or RNA-seq data?</B></font></A></H4>

Yes, you can use VarScan for any of these. The default settings are optimized for exome data, where one expects to have 
at least 10x or 20x coverage across targeted exons. By lowering the minimum coverage requirement (to perhaps 6x) one 
can call variants in lower-coverage data. Warning: the output files for WGS data may be quite large, and the runtimes longer,
since it will call 3+ million variants per genome. VarScan works for RNA-seq data as well, though this data tends to be 
noisier for variant calling due to RT errors, allele-specific expression, and alignment difficulties.

<BR><BR>

<H4><A NAME="usage-pooled-samples"><font CLASS="likeH3"><B>Does VarScan work for pooled samples?</B></font></A></H4>

Absolutely. It's simply a matter of setting appropriate input parameters, particularly 
<I>--min-coverage, --min-var-freq,</I> and <I>--p-value</I>.
In pooled data, you might specify a higher minimum coverage, a lower variant allele frequency 
(conservatively, 0.50 / the number of samples), and a less-stringent p-value to detect rare variants. 

<BR><BR>

<H4><A NAME="usage-validation"><font CLASS="likeH3"><B>How do I use VarScan for validation?</B></font></A></H4>

If validating germline SNPs/indels on an orthogonal sequencing platform, use the <I>pileup2cns</I> command, which will call 
consensus genotypes at all positions with sufficient coverage, rather than just the variants. This lets you determine 
sites that are refuted as wild-type. You might start with the following parameters:<BR>
<pre class="sequenceText">	
--min-coverage 20
--min-var-freq 0.08
--p-value 0.05
</pre>

If validating somatic SNPs/indels on an orthogonal sequncing platform, use the <I>somatic</I> command with 
<I>--validation</I> set to 1.
You might start with the following parameters:
<pre class="sequenceText">	
--min-coverage 20
--min-var-freq 0.08
--p-value 0.10
--somatic-p-value 0.05
--validation 1
</pre>


<BR><BR>

<H4><A NAME="usage-model-microbial"><font CLASS="likeH3"><B>Can I use VarScan on my model organism or microbial genome?</B></font></A></H4>

Of course. All you need is a BAM file of your reads aligned to a reference sequence. If you don't have a reference sequence
to which you might align reads, then no.

<BR><BR>

<H3>INPUT QUESTIONS</H3>
<H4><A NAME="input-what-format"><font CLASS="likeH3"><B>What input format should I give to VarScan?</B></font></A></H4>

VarScan expects its input in SAMtools pileup format, which is obtained from a BAM file via the samtools pileup command. 
For example:
<pre class="sequenceText">	
samtools pileup -f myReference.fasta myReads.bam >myPileup.pileup
java -jar VarScan.jar pileup2snp myPileup.pileup
</pre>
To save on disk space and I/O, you can also use a UNIX "pipe" command to forward the pileup output directly into VarScan:
<pre class="sequenceText">	
samtools pileup -f myRef.fasta myBam.bam | java -jar VarScan.jar pileup2snp
</pre>

<BR><BR>

<H4><A NAME="input-aligner-choice"><font CLASS="likeH3"><B>Which aligner should I use?</B></font></A></H4>

Great question! The choice of an aligner is an important one, and entire review articles have been written on the topic. 
For practical purposes, you should use an aligner whose output is (or can be converted to) a SAM/BAM file. I have heard 
that popular aligners for include BWA, Bowtie, and Novoalign for Illumina data, SHRiMP and BFAST for SOLiD data, and
SSAHA2 and BWA-SW for Roche/454 data.

<BR><BR>

<H4><A NAME="input-base-qualities"><font CLASS="likeH3"><B>Do I need Illumina or Phred scale base qualities?</B></font></A></H4>

VarScan expects Phred-scaled (Phred+33, also called "Sanger") base qualities, in which a score of 20 indicates a 1/1000 
probability of base error. By default, VarScan requires a minimum base quality of 15 or 20 (depending on the application), 
so this value should be adjusted appropriately if alternate base qualities are used.

<BR><BR>

<H4><A NAME="input-mapping-qualities"><font CLASS="likeH3"><B>What about mapping qualities?</B></font></A></H4>
<BR>

Many aligners provide a Phred-scaled quality value (mapping quality) for every read's alignment, which is correlated to 
the probability that the read is correctly mapped. A mapping quality of 0 typically indicates that the given read has 
many possible mapping locations of equal probability, and that the location given was chosen randomly. Thus, it's best 
to exclude reads with mapping quality of 0 from most downstream analyses. A minimum mapping quality of 10 is even better.
It's possible to apply this threshold to a BAM file using SAMtools, as follows:
<pre class="sequenceText">
samtools view -b -q 10 myBam.bam | samtools pileup -f myRef.fasta - 
| java -jar VarScan.jar pileup2snp
</pre>

<BR><BR>

<H3>OUTPUT QUESTIONS</H3>
<H4><A NAME="output-p-value"><font CLASS="likeH3"><B>What do the output columns mean?</B></font></A></H4>

By default, all VarScan output files should include headers. For detailed descriptions of the output columns 
and their meanings, see the "output" sections of the <A HREF="germline-calling.html#germline-output">germline</A> or 
<A HREF="somatic-calling.html#somatic-output">somatic</A> calling pages.

You can also specify VCF output, which is widely documented.<BR><BR>

<H4><A NAME="output-p-value"><font CLASS="likeH3"><B>How are the p-values calculated?</B></font></A></H4>

P-values are calculated using a Fisher's Exact Test on the read counts supporting reference and variant alleles. For details,
see the <A HREF="germline-calling.html">germline</A> or 
<A HREF="somatic-calling.html">somatic</A> calling pages.

<BR><BR>


<H4><A NAME="output-p-value-98"><font CLASS="likeH3"><B>Why are all of my p-values 0.98?</B></font></A></H4>

The p-value calculations are computationally expensive, so if the user doesn't specify a p-value threshold, VarScan 
skips the calculation in pileup2snp, pileup2indel, and pileup2cns. Instead, it inserts a dummy value of 0.98. To 
get p-values, set the <I>--p-value</I> parameter to something like 0.10, 0.05, or 0.01.

<BR><BR>

<H4><A NAME="output-filtering"><font CLASS="likeH3"><B>How should I filter the output files?</B></font></A></H4>

VarScan's default parameters aim to be sensitive when detecting variants, with the trade-off that it will report 
a lot of candidate variants, many of which will be false positives. 
These should be further filtered by coverage, variant allele frequency, strand 
representation, and p-value to isolate high-confidence calls. When performing the initial discovery, it is recommended
that you set <I>--strand-filter</I> to 1. After discovery, you should run VarScan <I>filter</I> or <I>somaticFilter</I> to further refine 
your variant calls. 

<BR><BR>

<H4><A NAME="output-confidence"><font CLASS="likeH3"><B>Does VarScan provide a confidence score similar to SAMtools?</B></font></A></H4>

It is possible to calculate a Phred-scaled confidence score from the p-value that VarScan provides:
<pre class="sequenceText">
$score = -10 * log10($p_value);
$score = 255 if($score > 255);
</pre>
This is done on a per-sample basis when <I>--vcf-output</I> is specified, since VCF format expects Phred-scaled confidence scores.

<BR><BR>

<H3>COMMON ISSUES, WARNINGS, AND ERRORS</H3>
<H4><A NAME="error-resetting-file"><font CLASS="likeH3"><B>Warnings about "resetting normal" or "resetting tumor" file</B></font></A></H4>

If you provide two separate input (pileup) files, VarScan attempts to use the chromosome and position to simultaneously parse both files and match them up. Doing so while accounting
for alphanumeric chromosome names is difficult, especially when there are chromosomes for which only one sample had coverage. VarScan does its best to ensure that no chromosomes 
are missed due to a sorting issue by resetting one sample's pileup file, to ensure that no chromosomes are missed due to sorting. When you have many chromosomes or contigs with 
coverage in just one sample, you'll see a lot of these warnings.
<BR /><BR />
The best way to address this simultaneous parsing issue is quite simple: <strong>provide VarScan with a two-sample MPILEUP file</strong> (normal and tumor) instead of two individual pileup files: 
<pre>
samtools mpileup -f reference.fasta -q 1 -B normal.bam tumor.bam >normal-tumor.mpileup 
java -jar VarScan.jar somatic normal-tumor.mpileup normal-tumor.varScan.output --mpileup 1
</pre>
In the mpileup file, SAMtools already does the chromosome and position matching-up, so there's no room for error. 
You'll also notice that I provided the -B parameter. This disables SAMtools BAQ computation, which is turned 
on by default in SAMtools mpileup, but (as the author admits) occasionally too stringent for variant calling.


<H4><A NAME="read-counts-different"><font CLASS="likeH3"><B>Read counts from VarScan are different from SAMtools/IGV counts</B></font></A></H4>

By default, SAMtools and IGV show and count all bases at a given position, regardless of base quality. In contrast, VarScan requires that bases meet the minimum Phred 
quality score (default 15 for most commands) to count them for things like read counts (reads1, reads2) and to compute variant allele frequency. However, when VarScan
reports the depth (such as in the DP field of VCF output), it reports SAMtools raw depth. To get VarScan read counts to more closely match another tool, set 
use parameter --min-avg-qual 0. And use caution! Low-quality bases, with the occasional exception of BAQ penalties, should not be trusted.
<BR /><BR />

Also, VarScan reports variants on a biallelic basis. That is, for a given SNP call, the "reads1" column is the number of reference-supporting reads (RD), and the "reads2" column is 
the number of variant-supporting reads (AD). There may be additional reads at that position showing other bases (SNP or indel variants). If these other variants meet the calling 
criteria, they will be reported in their own line. If not, it may look like you have "missing" reads.

<BR><BR>
<BR><BR>

<!-- InstanceEndEditable -->
</div>

<div style="clear: both;"> </div>

</div>

<div id="footer">
Copyright Â© 2009-2013 by <A HREF="http://www.wustl.edu" TITLE="About Dan Koboldt" TARGET="_blank">Washington University in St. Louis</A>. Design by <a href="http://www.free-css-templates.com/" TARGET="_blank">CSS Templates</a>
</div>

</div>

</body>
<!-- InstanceEnd --></html>